platform/darwin/src/MGLMapCamera.mm

```diff
@@ -102,6 +132,31 @@ - (id)copyWithZone:(nullable NSZone *)zone
                                                                heading:_heading];
 }
 
++ (NSSet<NSString *> *)keyPathsForValuesAffectingViewingDistance {
+    return [NSSet setWithObjects:@"altitude", @"pitch", nil];
+}
+
+- (CLLocationDistance)viewingDistance {
+    CGFloat pitch = self.pitch;
+    CLLocationDirection groundAngle = 90;
+    // Angle at the viewpoint formed by the shortest straight lines to the
+    // ground and center coordinate.
+    CLLocationDirection eyeAngle = 360 - groundAngle - pitch;
+    CLLocationDistance altitude = self.altitude;
+    // Solution of triangles given angle-side-angle.
+    return altitude * sin(groundAngle) / (sin(groundAngle) * cos(eyeAngle) + sin(eyeAngle) * cos(groundAngle));
```

Comment:
groundAngle = PI/2, sin(PI/2) = 1 and cos(PI/2) = 0.
This equation is effectively altitude / cos(eyeAngle) which produces a negative number.
I think eyeAngle is meant to be the difference between the pitch value and 90.

Try this:
```
- (CLLocationDistance)viewingDistance {
    CLLocationDirection eyeAngleFromGround = 90 - self.pitch;
    CGFloat eyeAngleFromGroundRadians = eyeAngleFromGround * M_PI_2 / 180;
    CLLocationDistance altitude = self.altitude;
    return altitude / sin(eyeAngleFromGroundRadians)
}
```